### PR TITLE
SDKv12 migration containerregistry

### PR DESCRIPTION
**Pull request by mbfrahry**
This PR migrates containerregistry over to the new sdk. It also fixes the migrate container registry test that failed due to a nil context 

```
make testacc TEST=./azurerm TESTARGS="-run=TestAccAzureRMContainerRegistry"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./azurerm -v -run=TestAccAzureRMContainerRegistry -timeout 180m
=== RUN   TestAccAzureRMContainerRegistry_importBasicClassic
--- PASS: TestAccAzureRMContainerRegistry_importBasicClassic (82.98s)
=== RUN   TestAccAzureRMContainerRegistry_importBasicBasic
--- PASS: TestAccAzureRMContainerRegistry_importBasicBasic (55.96s)
=== RUN   TestAccAzureRMContainerRegistry_importBasicManagedStandard
--- PASS: TestAccAzureRMContainerRegistry_importBasicManagedStandard (40.99s)
=== RUN   TestAccAzureRMContainerRegistry_importBasicManagedPremium
--- PASS: TestAccAzureRMContainerRegistry_importBasicManagedPremium (55.85s)
=== RUN   TestAccAzureRMContainerRegistry_importComplete
--- PASS: TestAccAzureRMContainerRegistry_importComplete (66.05s)
=== RUN   TestAccAzureRMContainerRegistryMigrateState
--- PASS: TestAccAzureRMContainerRegistryMigrateState (23.69s)
```

### CODE REVIEW

File: azurerm/resource_arm_container_registry_migrate_test.go [hunk 1]
```diff
@@ -24,6 +24,8 @@ func TestAccAzureRMContainerRegistryMigrateState(t *testing.T) {
 		return
 	}
 
+	client.StopContext = testAccProvider.StopContext()
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_container_registry_migrate_test.go]: given this is set in `config.go` - I don't believe this is needed?

Comment (tombuildsstuff) [reply azurerm/resource_arm_container_registry_migrate_test.go]: Taking a look into this - it is
